internal/ad/ad.go

```diff
@@ -448,16 +447,28 @@ func (ad *AD) parseGPOs(ctx context.Context, gpos []gpo, objectClass ObjectClass
 
 			log.Debugf(ctx, "Parsing GPO %q", name)
 
-			class := "User"
+			// We need to consider the uppercase version of the name as well,
+			// which could occur in some of the default GPOs such as Default
+			// Domain Policy.
+			classes := []string{"User", "USER"}
 			if objectClass == ComputerObject {
-				class = "Machine"
+				classes = []string{"Machine", "MACHINE"}
 			}
-			f, err := os.Open(filepath.Join(ad.sysvolCacheDir, "Policies", filepath.Base(url), class, "Registry.pol"))
-			if errors.Is(err, fs.ErrExist) {
-				return err
-			} else if errors.Is(err, fs.ErrNotExist) {
+
+			var f *os.File
+			var err error
+			for _, class := range classes {
+				f, err = os.Open(filepath.Join(ad.sysvolCacheDir, "Policies", filepath.Base(url), class, "Registry.pol"))
+				if err == nil {
+					break
+				}
+			}
+
+			if errors.Is(err, fs.ErrNotExist) {
```

Comment:
The issue is that we are returning the last error, not the first one. So, the error message displayed to the user will be with the capitalized letter path which is more an edge case than the regular one and may create confusion to them, what do you think?